packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -159,15 +159,16 @@ typedef VMServiceConnector = Future<FlutterVmService> Function(Uri httpUri, {
 });
 
 /// A connection to the Dart VM Service.
-vm_service.VmService setUpVmService(
+Future<vm_service.VmService> setUpVmService(
```

Comment:
by awaiting all of the registration futures we make the unit tests avoid the need for `await null` and other hacks